src/rpc/blockchain.cpp

```diff
@@ -2539,34 +2526,58 @@ UniValue createmrcrequest(const UniValue& params, const bool fHelp) {
         throw runtime_error("MRC request creation failed.");
     }
 
-    if (!dry_run) {
-        if (!force && reward == fee) {
-            throw runtime_error("MRC request is in zero payout interval.");
+    if (provided_fee != 0) {
+        if (!force) {
+            if (provided_fee < fee) {
+                throw runtime_error("Provided fee lower than required.");
+            }
+
+            if (provided_fee > mrc.m_research_subsidy) {
+                throw runtime_error("Provided fee higher than subsidy.");
+            }
         }
 
-        if (!force) {
-            for (const auto& [_, tx] : mempool.mapTx) {
-                for (const auto& contract: tx.GetContracts()) {
-                    if (contract.m_type == GRC::ContractType::MRC) {
-                        GRC::MRC mempool_mrc = contract.CopyPayloadAs<GRC::MRC>();
-
-                        GRC::CpidOption mempool_mrc_cpid = mempool_mrc.m_mining_id.TryCpid();
-                        GRC::CpidOption mrc_cpid = mrc.m_mining_id.TryCpid();
-
-                        if (mempool_mrc_cpid && mrc_cpid && *mempool_mrc_cpid == *mrc_cpid) {
-                            throw runtime_error("Oustanding MRC request already present in the mempool for CPID.");
-                        } // uniqueness test for cpid
-                    } // match to mrc contract type
-                } // contract iterator
-            } // mempool transaction iterator
+        mrc.m_fee = provided_fee;
+    }
+
+    if (!dry_run && !force && reward == fee) {
+        throw runtime_error("MRC request is in zero payout interval.");
+    }
+
+    int pos{0};
+    bool found{false};
+    for (const auto& [_, tx] : mempool.mapTx) {
+        for (const auto& contract: tx.GetContracts()) {
+            if (contract.m_type == GRC::ContractType::MRC) {
+                GRC::MRC mempool_mrc = contract.CopyPayloadAs<GRC::MRC>();
+
+                found |= mrc.m_mining_id == mempool_mrc.m_mining_id;
+                pos += mempool_mrc.m_fee >= mrc.m_fee;
+            } // match to mrc contract type
+        } // contract iterator
+    } // mempool transaction iterator
+
+    if (!dry_run && !force) {
+        if (found) {
+            throw runtime_error("Oustanding MRC request already present in the mempool for CPID.");
         }
 
+        if (pos >= static_cast<int>(GetMRCOutputLimit(pindex->nVersion, false))) {
+            throw runtime_error("MRC request queue is full.");
+        }
+    }
```

Comment:
Let's give the user a little more info here. Perhaps an error showing the fee required to get a slot at the end, and the fee currently required to go to the head?